met/src/tools/core/grid_stat/grid_stat.cc

```diff
@@ -185,6 +187,15 @@ static bool read_data_plane(VarInfo* info, DataPlane& dp, Met2dDataFile* mtddf,
 
 int main(int argc, char *argv[]) {
 
+#ifdef _OPENMP
+#pragma omp parallel
+#pragma omp single
+  {
+    // Report number of threads if compiled with OpenMP
+    std::cout << "Running on " << omp_get_num_threads() << " threads.\n";
```

Comment:
```suggestion
    mlog << Debug(2) << "Running on " << omp_get_num_threads() << " threads.\n";
```
Recommend routing log message through the global logging class instance (mlog) instead of cout.

Reply:
Happy to do.  I'd first like to check that's indeed what we want to do, though.  My thinking is: it's valuable to have information about the runtime configuration in the main "stdout" file, so that we can cross-reference it with, say, execution time or even just determine whether the executable contains OpenMP.  I think it needs to be more than a debug message.  With those things in mind, what would be the best way to code it?

Reply:
@mo-mglover very, very few of the log messages written by MET go to stdout. One exception is a rather annoying message from GNU Scientific Library when we create a random number generator and can't get it to write anywhere other than stdout. Any error messages occurring before the global logger is instantiated must also be written to stdout. But in general, if we can log via the logger, we do. The logging class does in fact write to stdout, as well as redirect log messages to a log file, if so requested using the "-log filename" command line option. Each call to the logger starts with either Error, Warning, or Debug(n) where n is the logging level. If it's useful, we could add other categories of log messages for performance. But we'd need to decide the rules for if/when those log messages should be enabled/disabled. Right now the user can set "-v log_level" on the command line (with a default of 2). If the log message level in the code <= log_level, it gets printed to stdout and the log file (if one is specified).